.github/workflows/reusable_build.yml

```diff
@@ -83,31 +83,21 @@ jobs:
           then
               BUILD_DEVICE_NAME="$(echo ${{ matrix.device }} | sed 's/nanosp/nanosplus/')" && \
               cd ${{ needs.call_get_app_metadata.outputs.build_directory }}
-              # Get the list of crates to update from cargo tree and filter only the required crates
+              # Update the Rust SDK crates
               all_crates=$(cargo +$RUST_NIGHTLY tree --depth 1 --prefix none | awk '{print $1}' | sort -u)
-              crates=()
               for crate in "ledger_secure_sdk_sys" "ledger_device_sdk" "include_gif"; do
-                  if echo "$all_crates" | grep -q "^$crate$"; then
-                      crates+=("$crate")
-                  fi
-              done
-              for crate in "${crates[@]}"; do
-                  tree_version_curr=$(cargo +$RUST_NIGHTLY tree --depth 1 --prefix none | grep $crate | awk '{print $2}' | tr -d 'v')
-                  echo "==> Try to update $crate"
+                if echo "$all_crates" | grep -q "^$crate$"; then
                   cargo +$RUST_NIGHTLY update $crate
-                  tree_version_new=$(cargo +$RUST_NIGHTLY tree --depth 1 --prefix none | grep $crate | awk '{print $2}' | tr -d 'v')
-                  if [[ $tree_version_curr != $tree_version_new ]]; then
-                      echo "$crate : version is $tree_version_curr whereas $tree_version_new is available; please update !"
-                      echo "Don't forget to activate dependabot in your repository settings to automatically update the Rust SDK dependencies"
-                      echo "see https://github.com/LedgerHQ/app-boilerplate-rust/blob/main/.github/dependabot.yml"
-                      exit 1
-                  else
-                      latest_version=$(cargo +$RUST_NIGHTLY search $crate | grep "^$crate" | awk '{print $3}' | tr -d '"')
-                      if [[ $latest_version != $tree_version_curr ]]; then
-                        echo "$crate : version is enforced to $tree_version_curr whereas $latest_version is available; please update !"
-                        exit 1
-                      fi
-                  fi
+                fi
+              done
+              # Check if update has run successfully
+              for crate in "ledger_secure_sdk_sys" "ledger_device_sdk" "include_gif"; do
+                curr_version=$(cargo +$RUST_NIGHTLY tree -p $crate | grep -oE "$crate v[0-9]+\.[0-9]+\.[0-9]+" | awk '{print $2}' | sed 's/^v//')
+                latest_version=$(cargo search $crate --limit 1 | grep -oE "$crate = \"[0-9]+\.[0-9]+\.[0-9]+\"" | awk -F '"' '{print $2}')
+                if [[ "$curr_version" != "$latest_version" ]]; then
+                  echo "Update for $crate failed: current version: $curr_version, latest version: $latest_version"
+                  exit 1
+                fi
```

Comment:
Is it relevant to check the version after the update here?
Just like we don't check package version after an `apt install`, I hope we can trust `cargo` to behave correctly.
And if we don't, is it the role of this CI to stop the build?

Reply:
yes indeed....